### PR TITLE
Replace fmpp-maven-plugin with fmpp Ant Task to avoid pulling the vulnerable log4j jars

### PR DESCRIPTION
**Pull request by chrajeshbabu**
Solve #14011

### CODE REVIEW

File: pinot-common/pom.xml [hunk 1]
```diff
@@ -301,6 +303,10 @@
       <artifactId>equalsverifier</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (rfscholte) [inline pinot-common/pom.xml:306]: this makes it a project dependency, while it is only used by the plugin. According to https://maven.apache.org/plugins/maven-antrun-plugin/examples/customTasks.html you should add the dependency to the plugin .

File: pinot-common/pom.xml [hunk 2]
```diff
@@ -100,22 +100,31 @@
...
+            <goals>
+              <goal>run</goal>
+            </goals>
           </execution>
         </executions>
+        <dependencies>
+          <dependency>
+            <groupId>net.sourceforge.fmpp</groupId>
+            <artifactId>fmpp</artifactId>
+            <version>${fmpp.version}</version>
```
Comment (Jackie-Jiang) [inline pinot-common/pom.xml:125]: Can we pull this up to the parent pom dependency management?

Comment (rfscholte) [reply pinot-common/pom.xml:125]: It is only used in this place, but looking at the pinot-project where every version is managed in the root pom, that would make sense. (personally I would also put them in alphabetic order, but that's something for another PR)

Comment (chrajeshbabu) [reply pinot-common/pom.xml]: Rebased PR by fixing the conflicts with latest code. Added the dependency at the root pom as agreed and follow the similar dependency management.

Comment (chrajeshbabu) [reply pinot-common/pom.xml]: @rfscholte  https://github.com/Ekryd/sortpom is the good candidate for managing the clean pom and sorting the dependencies in order. If you have not started will take care of it and raise PR for it.

Comment (chrajeshbabu) [reply pinot-common/pom.xml]: Thanks for reviews @rfscholte @Jackie-Jiang.

**Comment by chrajeshbabu** [comment]
Test case failures may not be relevant to the patch.

**Comment by chrajeshbabu** [comment]
Rebased to latest code @Jackie-Jiang.

**Comment by Jackie-Jiang** [comment]
Seems the tests are constantly failing (I've rerun them several times). Can you take a look?

**Comment by chrajeshbabu** [comment]
> Seems the tests are constantly failing (I've rerun them several times). Can you take a look?

@Jackie-Jiang sure will look into them.

**Comment by Jackie-Jiang** [comment]
Seems the failure is caused by #14237 and not related to this PR. Will take a look and get back

**Comment by Jackie-Jiang** [comment]
@chrajeshbabu Can you rebase this to the current master? The flaky test has been fixed in the master

**Comment by chrajeshbabu** [comment]
> @chrajeshbabu Can you rebase this to the current master? The flaky test has been fixed in the master

Will do in sometime @Jackie-Jiang . Thanks

**Comment by chrajeshbabu** [comment]
> > @chrajeshbabu Can you rebase this to the current master? The flaky test has been fixed in the master
> 
> Will do in sometime @Jackie-Jiang . Thanks

Rebased.

**Review by Jackie-Jiang — APPROVED** [review]

**Comment by chrajeshbabu** [comment]
Thank you @Jackie-Jiang for merging.

**Pull request merged**